### PR TITLE
Uploading, deleting, and downloading images uploaded to sanity.io

### LINKED ISSUE #6956: Uploading, deleting, and downloading images uploaded to sanity.io
#### Proposed title of article
 Uploading, deleting, and downloading images uploaded to sanity.io

#### Proposed article introduction
Sanity.io is a structured content platform. With the help of sanity.io, you can manage text, images, and other media with APIs. Also, you can use the open-source single-page application Sanity Studio to quickly set up an editing environment that you can customize.

In this tutorial, the developer will learn how all this information can be managed on the frontend (website) instead of using Sanity Studio. Using GROQ, we will upload images to sanity.io (Content Lake Database) and retrieve them to be viewed on our website. Afterward, we will implement the delete and download features.

#### Key takeaways 
-  Working with GROQ 
-  Uploading images on the website
-  Retrieving images from Sanity.io
-  Deleting images on the website
-  How can images be downloaded on the website

#### Article quality
The developer will learn both basic/beginner and advanced knowledge on how to use sanity with react js. The developer will also learn how to upload images from the website instead of using sanity cms. The developer will also learn how to use GROQ's sanity query language.

#### References
N/A

### PR DESCRIPTION
**Pull request by ben-cdm**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6956

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Plagiarism check complete

File: content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+
+- As a Content Management System (CMS), the administrator usually has all the permissions, e.g., managing text, images, and any other type of information.
+- Only to store data and perform updates on the website.
+
+In this blog, we will build a simple web application using sanity.io to store data. Then, we will upload the photos to the database and retrieve them using GROQ. Functionality such as uploading, deleting, and downloading uploaded images can be integrated into the website.
+
+Let us begin!
+
+### Prerequisites
+
```
Comment (mohamedgh16) [inline content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md:15]: Remove extra space lines after each header.
implement this in the entire article.

File: content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+5.  [React Masonry CSS](https://www.npmjs.com/package/react-masonry-css)
+6.  [React-icons](https://react-icons.github.io/react-icons)
+7.  [React-router-dom](https://www.npmjs.com/package/react-router-dom)
+
+### Installation
+
+1. [Sanity.io](https://sanity.io/)
+
+To install sanity.io, follow the steps below:
+
```
Comment (mohamedgh16) [inline content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md:35]: Remove any unnecessary space lines.

File: content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+```bash
+sanity init
+```
+
+For the Sanity API to set up the best project structure for us, select the following in the terminal dialog:
+
+- Login using your email.
+- Create a new project.
+- Choose the default dataset configuration by typing y, then hit enter.
+- Select project template. Then, choose a clean project with no predefined schemas.
```
Comment (mohamedgh16) [inline content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md:64]: Fix the spelling errors here.

File: content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md [hunk 4]
```diff
@@ -0,0 +1,686 @@
...
+
+7. [React-router-dom](https://www.npmjs.com/package/react-router-dom)
+
+```bash
+npm i react-router-dom
+```
+
+After installation, you should have a file structure as follows:
+
+![sanity project file structure](/engineering-education/articles-uploading-deleting-and-dowloading-images-uploaded-to-sanity/filestructure.PNG)
```
Comment (mohamedgh16) [inline content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md:141]: Downloading.

File: content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md [hunk 5]
```diff
@@ -0,0 +1,686 @@
...
+
+Ensure you are in the backend directory to view the output design. Then type in the terminal:
+
+```bash
+sanity start
+```
+
+Your sanity.io CMS should look like this:
+
+![sanity.io CMS](/engineering-education/articles-uploading-deleting-and-dowloading-images-uploaded-to-sanity/sanitycms.png)
```
Comment (mohamedgh16) [inline content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md:199]: Downloading.

File: content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md [hunk 6]
```diff
@@ -0,0 +1,686 @@
...
+
+### Working on the Web application
+
+### Step 1: Working with react.js
```
Comment (mohamedgh16) [inline content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md:203]: Use an extra hash (#) to variate the main header from the subheader.

File: content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md [hunk 7]
```diff
@@ -0,0 +1,686 @@
...
+export default uploadImage
+```
+
+Now, try to upload an image from your website. After inputting the image, you should have something like this:
+
+![sanity.io CMS after upload](/engineering-education/articles-uploading-deleting-and-dowloading-images-uploaded-to-sanity/afterinput.png)
+
+At your sanity CMS desk, you should have the photo.
+
+![sanity.io CMS after upload](/engineering-education/articles-uploading-deleting-and-dowloading-images-uploaded-to-sanity/sanitycmsafterupload.png)
```
Comment (mohamedgh16) [inline content/articles/uploading-deleting-and-downloading-images-uploaded-to-sanity/index.md:485]: Fix the word downloading in the link.

**Review by ghost — COMMENTED** [review]
Great article @ben-cdm 
Implement the changes so can move to tech review.

**Comment by ben-cdm** [comment]
Hey @mohamedgh16 I have made the minor changes you mentioned. Please consider checking. Thank you.

**Comment by ben-cdm** [comment]
Hey @mohamedgh16 I have made the minor changes you mentioned. Please consider checking. Thank you.

**Comment by hectorkambow** [comment]
@ben-cdm lets be sure we are doing a full review of our own work before asking for a review. 👍 

to avoid any potential closures 👍

**Comment by hectorkambow** [comment]
tech-related issues, grammar etc.

**Comment by hectorkambow** [comment]
cc: 
@mohamedgh16

**Comment by mohamedgh16** [comment]
@ben-cdm Great work, please consider the correction to minimize revision next time👍.

**Comment by hectorkambow** [comment]
We will be reviewing this (& publishing if it is ready) ASAP.

**Comment by hectorkambow** [comment]
👏  

Well done! 

If you have not done so already - consider adding a skill section to your author profile. 

This will go a long way with potential employers.

**Pull request merged**